core/src/datayoga_core/blocks/files/read_csv/block.py

```diff
@@ -24,12 +26,29 @@ def init(self, context: Optional[Context] = None):
 
         logger.debug(f"file: {self.file}")
         self.batch_size = self.properties.get("batch_size", 1000)
+        self.fields = self.properties.get("fields")
+        self.skip = self.properties.get("skip", 0)
+        self.delimiter = self.properties.get("delimiter", ",")
+        self.quotechar = self.properties.get("quotechar", "\"")
 
     async def produce(self) -> AsyncGenerator[List[Message], None]:
         logger.debug("Reading CSV")
 
         with open(self.file, "r") as read_obj:
-            records = list(DictReader(read_obj))
+            reader = DictReader(read_obj, fieldnames=self.fields, delimiter=self.delimiter, quotechar=self.quotechar)
+            counter = iter(count())
 
-        for i, record in enumerate(records):
-            yield [{self.MSG_ID_FIELD: str(i), **record}]
+            for _ in range(self.skip):
+                with suppress(StopIteration):
+                    next(reader)
+
+            while True:
+                sliced = islice(reader, self.batch_size)
+                records = [{self.MSG_ID_FIELD: str(next(counter)), **record} for record in sliced]
+
+                logger.debug(f"Producing {len(records)} records")
```

Comment:
wouldn't we get an error here if records in falsie? so probably the check in line 51 should come first.